rust/main/utils/run-locally/src/main.rs

```diff
@@ -483,6 +484,15 @@ fn main() -> ExitCode {
     // give things a chance to fully start.
     sleep(Duration::from_secs(10));
 
+    // test retry request
+    let retry_req = std::thread::spawn(server::run_retry_request);
```

Comment:
do you need to `std::thread::spawn` / join? afaict run_retry_request isn't async and does some tokio orchestration so it could be called directly from a sync place like here

Reply:
Good call, I originally had this because I was running into errors that I thought were tokio runtime errors.
I just ran it again without the thread spawn and it worked so I will remove it